Week02/Assignment/afilbrun_01.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jul 18 14:16:58 2018
+
+@author: amandl
+"""
+
+# import python libraries and provided data 
+
+import numpy as np
+import scipy as sp
+import matplotlib.pyplot as plt
+
+x = np.array([ 1.,   1.5,  2.,   2.5,  3.,   3.5,  4.,   4.5,  5.,   5.5,  6.,   6.5,  7.,   7.5, 8.,   8.5,  9.,   9.5, 10. ])
+y = np.array([3.43, 4.94, 6.45, 9.22, 6.32, 6.11, 4.63, 8.95, 7.8, 8.35, 11.45, 14.71, 11.97, 12.46, 17.42, 17.0, 15.45, 19.15, 20.86])
+
+# 1 & 2 - Linear and Cubic Curve Fitting
+
+from scipy.stats import linregress
+slope, intercept, r_value, p_value, std_err = linregress(x,y)
+
+coefficients = np.polyfit(x, y, 3)
+pfit = np.poly1d(coefficients)
+
+# 3- Calculate area under the curve
+
+area,err = sp.integrate.quad(pfit,x[0],x[-1])
+print("%.2f" % area)
+
+# 4 & 5- Generate Figure
+
+fig = plt.figure()
+axes = fig.add_axes([0.1, 0.1, 0.8, 0.8])
+
+axes.scatter(x,y,c = 'k') # plot data
+y = y.astype('float64')
+plt.plot(x,x*slope + intercept) # plot linear fit
+plt.plot(x,pfit(x),'r') # plot cubic fit
+
+
+axes.set_title('Linear and Cubic Fits of Provided Data')
+axes.set_xlabel('x values')
+axes.set_ylabel('y values')
+axes.legend(['linear fit','cubic fit'])
+
+axes.text(5,5,'Area Under Cubic Curve: ' + "%.2f" % area)
+
+fig.savefig('afilbrun_plot.pdf')
+
+# 6-  Bayesian values. 
+def BIC(y, yhat, k, weight = 1):
+    err = y - yhat
+    sigma = np.std(np.real(err))
+    n = len(y)
+    B = n*np.log(sigma**2) + weight*k*np.log(n)
+    return B
+
+coefficients2 = np.polyfit(x, y, 1) 
+pfit2 = np.poly1d(coefficients2)
+
+h1 = BIC(y,pfit2(x),1)
+h2 = BIC(y,pfit(x),3)
```

Comment:
Here you're calculating the BIC of a linear model as if it has one parameter, it actually has 2 (slope and intercept). Similarly a cubic model has 4 parameters. I know this is confusing since I made this mistake in the recitation itself.